components/datetime/src/timezone.rs

```diff
@@ -0,0 +1,331 @@
+        let mut mz_generic_short: Option<Cow<MetaZoneGenericNamesShortV1>> = None;
+        let mut mz_specific_long: Option<Cow<MetaZoneSpecificNamesLongV1>> = None;
+        let mut mz_specific_short: Option<Cow<MetaZoneSpecificNamesShortV1>> = None;
+
+        for resource_key in required_time_zone_keys(&pattern)? {
+            if resource_key.eq(&provider::key::TIMEZONE_EXEMPLAR_CITIES_V1) {
+                load_resource(&locale, resource_key, &mut exemplar_cities, zone_provider)?;
+            } else if resource_key.eq(&provider::key::TIMEZONE_GENERIC_NAMES_LONG_V1) {
+                load_resource(&locale, resource_key, &mut mz_generic_long, zone_provider)?;
+            } else if resource_key.eq(&provider::key::TIMEZONE_GENERIC_NAMES_SHORT_V1) {
+                load_resource(&locale, resource_key, &mut mz_generic_short, zone_provider)?;
+            } else if resource_key.eq(&provider::key::TIMEZONE_SPECIFIC_NAMES_SHORT_V1) {
+                load_resource(&locale, resource_key, &mut mz_specific_short, zone_provider)?;
+            } else if resource_key.eq(&provider::key::TIMEZONE_SPECIFIC_NAMES_LONG_V1) {
+                load_resource(&locale, resource_key, &mut mz_specific_long, zone_provider)?;
+            }
+        }
+
+        Ok(Self {
+            pattern,
+            zone_formats,
+            exemplar_cities,
+            mz_generic_long,
+            mz_generic_short,
+            mz_specific_long,
+            mz_specific_short,
+        })
+    }
+
+    pub fn format<'s: 'd, T>(&'s self, value: &'s T) -> FormattedTimeZone<'s, T>
+    where
+        T: TimeZoneInput,
+    {
+        FormattedTimeZone {
+            time_zone_format: self,
+            time_zone: value,
+        }
+    }
+
+    pub fn format_to_write(
+        &self,
+        w: &mut impl std::fmt::Write,
+        value: &impl TimeZoneInput,
+    ) -> std::fmt::Result {
+        timezone::write_pattern(self, value, w).map_err(|_| std::fmt::Error)
+    }
+
+    pub fn format_to_string(&self, value: &impl TimeZoneInput) -> String {
+        let mut s = String::new();
+        self.format_to_write(&mut s, value)
+            .expect("Failed to write to a String.");
+        s
+    }
+
+    /// Returns the time zone in generic location format as defined by the UTS-35 spec.
+    /// e.g. France Time
+    /// https://unicode.org/reports/tr35/tr35-dates.html#Time_Zone_Format_Terminology
+    pub(super) fn generic_location_format(
+        &self,
+        time_zone: &impl TimeZoneInput,
+    ) -> Option<Cow<str>> {
+        self.exemplar_city(time_zone)
+            .map(|location| Cow::Owned(self.zone_formats.region_format.replace("{0}", &location)))
+    }
+
+    /// Returns the time zone in short generic non-location format as defined by the UTS-35 spec.
+    /// e.g. PT
+    /// https://unicode.org/reports/tr35/tr35-dates.html#Time_Zone_Format_Terminology
+    pub(super) fn short_generic_non_location_format(
```

Comment:
Suggestion (optional): These methods are just fetching cells from data structs.  They would make more sense somewhere in the `crate::provider::helpers` module, for example, or perhaps even methods on the data struct itself.  You should be able to change the body of this function to something like

```rust
time_zone.metazone_id().map(|mz| self.mz_generic_short.get_name_for_id(mz))
```

Actually, that's not a whole lot cleaner than what you have now.  So feel free to ignore this suggestion.